resources/models.py

```diff
@@ -252,6 +263,49 @@ def __str__(self):
         return self.safe_translation_getter("name", super().__str__())
 
 
+class AbstractAreaMap(UUIDModel):
+    map_file = models.FileField(
+        upload_to=get_map_resource_media_folder,
+        storage=OverwriteStorage(),
+        verbose_name=_("map file"),
+        null=True,
+        blank=True,
+    )
+
+    class Meta:
+        abstract = True
+
+
+class HarborMap(AbstractAreaMap):
+    harbor = models.ForeignKey(
+        Harbor,
+        null=True,
+        blank=True,
+        verbose_name=_("harbor"),
+        related_name="harbor_maps",
+        on_delete=models.CASCADE,
+    )
+
+
+class WinterStorageAreaMap(AbstractAreaMap):
+    winter_storage_area = models.ForeignKey(
+        WinterStorageArea,
+        null=True,
+        blank=True,
+        verbose_name=_("winter storage area"),
+        related_name="winter_storage_area_maps",
+        on_delete=models.CASCADE,
+    )
+
+
+@receiver(post_delete, sender=HarborMap)
+@receiver(post_delete, sender=WinterStorageAreaMap)
+def delete_map_file_handler(sender, instance, **kwargs):
+    if instance.map_file:
+        file_path = Path(instance.map_file.path)
+        shutil.rmtree(file_path.parent)
```

Comment:
this deletes the entire parent folder? it should just delete the file.

Reply:
AHHH, right. I misunderstood how the media paths/files were being handled, but you're right.